packages/application-config/src/transformers.ts

```diff
@@ -19,16 +22,49 @@ const computeUriPath = (uriPath: string, entryPointUriPath: string) => {
   return `${entryPointUriPath}/${uriPath}`;
 };
 
+type TResourceAccessPermissionName<PermissionName extends string = ''> =
+  | `view${Capitalize<PermissionName>}`
+  | `manage${Capitalize<PermissionName>}`;
+
+type TImplicitCustomApplicationResourceAccesses<
+  PermissionName extends string = ''
+> = Record<TResourceAccessPermissionName<PermissionName>, string[]>;
+
 function transformCustomApplicationConfigToData(
   appConfig: JSONSchemaForCustomApplicationConfigurationFiles
 ): CustomApplicationData {
   validateEntryPointUriPath(appConfig);
   validateSubmenuLinks(appConfig);
 
+  const additionalResourceAccessKeyToOauthScopeMap = (
+    appConfig.additionalOAuthScopes || []
+  ).reduce(
+    (previousOauthScope, { name, view, manage }) => ({
+      ...previousOauthScope,
+      [`view${formatEntryPointUriPathToResourceAccessKey(name)}`]: view,
+      [`manage${formatEntryPointUriPathToResourceAccessKey(name)}`]: manage,
+    }),
+    {} as TImplicitCustomApplicationResourceAccesses<PermissionName>
+  );
+
+  const additionalPermissionNames =
+    appConfig.additionalOAuthScopes?.map(({ name }) => name) || [];
+
   const permissionKeys = entryPointUriPathToResourceAccesses(
-    appConfig.entryPointUriPath
+    appConfig.entryPointUriPath,
+    additionalPermissionNames
   );
 
+  const additionalPermissions = (
+    Object.keys(
+      additionalResourceAccessKeyToOauthScopeMap || []
+    ) as TResourceAccessPermissionName<PermissionName>[]
+  ).map((additionalResourceAccessKey) => ({
+    name: permissionKeys[additionalResourceAccessKey],
+    oAuthScopes:
+      additionalResourceAccessKeyToOauthScopeMap[additionalResourceAccessKey],
+  }));
```

Comment:
Nit: to not pollute much the `transformCustomApplicationConfigToData` function, maybe extract the transformation logic for the additional permissions into a separate function?

Reply:
Updated [11ceeee](https://github.com/commercetools/merchant-center-application-kit/pull/2822/commits/11ceeee3115d57516d36fbd6193cacca1758ef99)